integration/test_utils.go

```diff
@@ -48,28 +51,61 @@ var (
 )
 
 func init() {
+	if err := ConnectDaemons(); err != nil {
+		glog.Exitf("Failed to connect daemons: %v", err)
+	}
+}
+
+// ConnectDaemons connect cri-containerd and containerd, and initialize the clients.
+func ConnectDaemons() error {
 	var err error
 	runtimeService, err = remote.NewRemoteRuntimeService(sock, timeout)
 	if err != nil {
-		glog.Exitf("Failed to create runtime service: %v", err)
+		return fmt.Errorf("failed to create runtime service: %v", err)
 	}
 	imageService, err = remote.NewRemoteImageService(sock, timeout)
 	if err != nil {
-		glog.Exitf("Failed to create image service: %v", err)
+		return fmt.Errorf("failed to create image service: %v", err)
+	}
+	// Since CRI grpc client doesn't have `WithBlock` specified, we
+	// need to check whether it is actually connected.
```

Comment:
what does it mean to have a runtimeService or imageService (grpc service) but it's not actually connected? Will it fail right away? should we do a sleep loop here to wait for it to be actually connected?

Reply:
is there some grpc api or option we're supposed to be using to demand an actual connection or err?

Reply:
`WithBlock` the problem is that we are not using that in Kubernetes today.

Reply:
Should we file an issue and track here?